pkg/controller/migmigration/task.go

```diff
@@ -486,3 +516,17 @@ func (t *Task) addErrors(errors []string) {
 		t.Errors = append(t.Errors, error)
 	}
 }
+
+// Get the current step.
+// Returns: name, n, total.
+func (t *Task) getStep() (string, int, int) {
+	n := -1
+	total := len(Step)
+	for i, step := range Step {
```

Comment:
Would making Step a map instead of a slice make this easier -- then instead of looping over the full slice to find the element number you could just use the map to find it directly on each getStep call. Not a huge deal, as this should work fine, too.

Reply:
The advantage of the list is it provides order, enumeration and total number of steps.
I considered building a Map but thinking that iterating over a slice containing ~20 items vs. the overhead of hashing the string for the map lookup is pretty equal.

Reply:
It's not a big deal either way. It probably looks more cumbersome than it is in practice, but I suspect there are golang style issues at work here, too, since go doesn't provide an `index` method on arrays/slices like other languages do.